api.go

```diff
@@ -650,6 +635,38 @@ func (r *Raft) restoreSnapshot() error {
 	return nil
 }
 
+func (r *Raft) tryRestoreSingleSnapshot(snapshot *SnapshotMeta) bool {
```

Comment:
wouldn't be better to make `tryRestoreSingleSnapshot` return an error and create the logger outside of it and log the error when it's returned?

Reply:
Maybe? I wanted to have a derived logger here (`Logger.With()`) so that all of the progress logs and the errors got the same set of bonus KV data logged.

Given that, I'd end up creating the logger outside of this method to pass in for progress in `fsmRestoreAndMeasure`, but returning an error that is immediately logged by the caller, which seemed differently strange.

Also the two current logs are slightly different today:
```
snapLogger.Error("failed to open snapshot", "error", err)
snapLogger.Error("failed to restore snapshot", "error", err)
```

and are carried over from the existing code. If these were changed to `fmt.Errorf` return values there'd be a slight change of output as we'd end up having to do `snapLogger.Error(err.Error())` instead, and we'd lose the "error" hclog attribute.

I don't mind inverting that logic if you think it is warranted.